src/Middleware/integrations/ordercloud.integrations.cardconnect/OrderCloudIntegrationsCardConnectService.cs

```diff
@@ -24,16 +25,22 @@ public class OrderCloudIntegrationsCardConnectConfig
         public string CadMerchantID { get; set; }
         public string EurMerchantID { get; set; }
     }
+    public enum AppEnvironment { Test, Staging, Production }
 
     public class OrderCloudIntegrationsCardConnectService : IOrderCloudIntegrationsCardConnectService
     {
         private readonly IFlurlClient _flurl;
+        private bool noAccountCredentials;
+        private AppEnvironment appEnvironment;
         public OrderCloudIntegrationsCardConnectConfig Config { get; }
 
-        public OrderCloudIntegrationsCardConnectService(OrderCloudIntegrationsCardConnectConfig config, IFlurlClientFactory flurlFactory)
+        public OrderCloudIntegrationsCardConnectService(OrderCloudIntegrationsCardConnectConfig config, string environment, IFlurlClientFactory flurlFactory)
         {
             Config = config;
-            _flurl = flurlFactory.Get($"https://{Config.Site}.{Config.BaseUrl}/");
+            // if no credentials are provided in Test and UAT, responses will be mocked.
+            noAccountCredentials = config == null || (config?.Authorization == null && config.AuthorizationCad == null) || (config?.Authorization == "" && config?.AuthorizationCad == "");
+            appEnvironment = (AppEnvironment)Enum.Parse(typeof(AppEnvironment), environment);
```

Comment:
Why don't you pass in the enum instead of the stringified version? Would save you having to parse it

Reply:
Since the integrations don't have the headstart models as a dependency (by design), I had to get to a primitive type and then cast it back to the integration's enum. I can also just match on string, but casting back to enum seemed like the least failure-prone solution.

Reply:
Oh good point. I forgot about that